packages/gatsby/src/commands/build.js

```diff
@@ -137,14 +142,11 @@ module.exports = async function build(program: BuildArgs) {
     `BOOTSTRAP_QUERY_RUNNING_FINISHED`
   )
 
-  await db.saveState()
-
   await waitUntilAllJobsComplete()
 
-  // we need to save it again to make sure our latest state has been saved
-  await db.saveState()
```

Comment:
Those `db.saveState()` calls need to be restored - we rely on this on those saves to minimize amount of work on subsequent runs (in case something crashes, etc)

Reply:
Done =)